src/timeMachine/actions/scriptEditorThunks.ts

```diff
@@ -0,0 +1,45 @@
+// Libraries
+import {Dispatch} from 'react'
+import {get} from 'lodash'
+
+// API
+import {FluxdocsArray, getFluxdocs} from 'src/client/fluxdocsdRoutes'
+
+// Actions
+import {notify} from 'src/shared/actions/notifications'
+import {getFluxPackagesFailed} from 'src/shared/copy/notifications/categories/alerts'
+
+// Types
+import {NotificationAction} from 'src/types'
+
+export type Action = SetGetFluxFunc
+
+interface SetGetFluxFunc {
+  type: 'GET_FLUX_DOCS'
+  payload: {data: FluxdocsArray}
+}
+
+export const setFluxFunc = (data): SetGetFluxFunc => ({
+  type: 'GET_FLUX_DOCS',
+  payload: {
+    data,
+  },
+})
```

Comment:
While this is fine for a single action, this pattern becomes very wordy as more functions are added, because every `action` needs an interface. There is another pattern that we use which doesn't require an interface for every action. It looks like this:

```
export const GET_FLUX_DOCS = 'GET_FLUX_DOCS'

export type Action = ReturnType<typeof setFluxFunc>

export const setFluxFunc = (data: FluxdocsArray) =>
  ({
    type: GET_FLUX_DOCS,
    payload: {
      data,
    },
  } as const)
```

Reply:
I can see how this can get out of hand very quickly.